### PR TITLE
Test simple tags in Time

### PR DESCRIPTION
**Pull request by kidder**
## Proposed changes

Add a test helper for testing simple tags.
Use it to test simple tags in Time directory.
Remove unnecessary name functions in simple Time tags.
Cleanup some documentation of simple Time tags.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [x] Code
- [x] Documentation
- [ ] Build system
- [x] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: tests/Unit/DataStructures/DataBox/TestHelpers.hpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+#pragma once
```
Comment (nilsdeppe) [inline tests/Unit/DataStructures/DataBox/TestHelpers.hpp:4]: blank line

File: tests/Unit/DataStructures/DataBox/TestHelpers.hpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+void test_simple_tag(const std::string& expected_name) {
+  static_assert(cpp17::is_base_of_v<::db::SimpleTag, Tag> and
+                    not cpp17::is_base_of_v<::db::ComputeTag, Tag>,
+                "A simple tag must be derived from "
+                "db::SimpleTag, but not db::ComputeTag");
+  static_assert(not cpp17::is_same_v<Tag, typename Tag::type>,
+                "A type cannot be its own tag.");
+  CHECK(::db::tag_name<Tag>() == expected_name);
+  if (detail::is_name_callable_v<Tag>) {
+    INFO("Do not define name for Tag '" << ::db::tag_name<Tag>() << "',");
```
Comment (nilsdeppe) [inline tests/Unit/DataStructures/DataBox/TestHelpers.hpp:31]: I'm worried that with using `INFO` here you will collect all the messages for all tags that have a `name()` function. Have you verified that the scoping applies correctly?

File: tests/Unit/DataStructures/DataBox/TestHelpers.hpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#pragma once
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+namespace test {
```
Comment (nilsdeppe) [inline tests/Unit/DataStructures/DataBox/TestHelpers.hpp]: why not `TestHelpers`?

Comment (kidder) [reply tests/Unit/DataStructures/DataBox/TestHelpers.hpp]: `test` is shorter 😛 I'm fine with `TestHelpers`, but then we should document it...

File: tests/Unit/DataStructures/DataBox/TestHelpers.hpp (hunk 2 above)
Comment (kidder) [reply tests/Unit/DataStructures/DataBox/TestHelpers.hpp:31]: Yes, I got one INFO per error message

File: tests/Unit/DataStructures/DataBox/TestHelpers.hpp (hunk 3 above)
Comment (nilsdeppe) [reply tests/Unit/DataStructures/DataBox/TestHelpers.hpp]: We have been using `TestHelpers` and did tell Will to move something else into a `TestHelpers` namespace :P

File: tests/Unit/DataStructures/DataBox/TestHelpers.hpp (hunk 2 above)
Comment (nilsdeppe) [reply tests/Unit/DataStructures/DataBox/TestHelpers.hpp:31]: Cool, cool :P

**Review by nilsvu — COMMENTED** [review]
Other than the namespace change, this LGTM. Note that the new test helper is essentially a protocol conformance test (plus a runtime check for the expected name).

**Comment by kidder** [comment]
yeah, I had that in mind when writing it...

File: tests/Unit/DataStructures/DataBox/TestHelpers.hpp (hunk 1 above)
Comment (nilsdeppe) [reply tests/Unit/DataStructures/DataBox/TestHelpers.hpp:4]: ping

**Review by nilsdeppe — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Comment by kidder** [comment]
@nilsleiffischer please look at as I added a commit

**Review by nilsvu — APPROVED** [review]

**Pull request merged**